python/cudf/cudf/dataframe/index.py

```diff
@@ -728,7 +728,7 @@ def __init__(self, values, name=None):
 
     @property
     def codes(self):
-        return self._values.codes
+        return self._values.cat().codes.to_array()
```

Comment:
We won't want this to automatically transfer from device to host. If we don't want this to return a Series because Pandas returns an array, we could perhaps return the underlying devicearray

Reply:
Changed this to return Series instead of a host array.